packages/mui-base/src/Badge/Badge.spec.tsx

```diff
@@ -26,24 +26,51 @@ const polymorphicComponentTest = () => {
       return <div />;
     };
 
+  const CustomRoot = function CustomRoot() {
+    return <div />;
+  };
+
   return (
     <div>
       {/* @ts-expect-error */}
       <Badge invalidProp={0} />
 
-      <Badge component="a" href="#" />
+      <Badge<'a'>
+        slots={{
+          root: 'a',
+        }}
+        href="#"
+      />
 
-      <Badge component={CustomComponent} stringProp="test" numberProp={0} />
-      {/* @ts-expect-error */}
-      <Badge component={CustomComponent} />
+      <Badge<typeof CustomComponent>
+        slots={{
+          root: CustomComponent,
+        }}
+        stringProp="test"
+        numberProp={0}
+      />
+
+      {/* @ts-expect-error onClick must be specified in the custom root component */}
+      <Badge<typeof CustomRoot> slots={{ root: CustomRoot }} onClick={() => {}} />
```

Comment:
IMO this test was pretty specific to a button. I don't think we need to have it in all components

Reply:
addressed